spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+
+describe OmniauthCallbacksController do
+
+  fixtures :users
+
+  describe '#login_dot_gov' do
+    before do
+      mock_user_auth('12345', 'test@gsa.gov')
+      request.env['omniauth.auth'] = OmniAuth.config.mock_auth[:login_dot_gov]
+      get :login_dot_gov
+    end
+
+    let(:user) { User.find_by(email: 'test@gsa.gov') }
+
+    it { is_expected.to redirect_to(admin_home_page_path) }
+
+    context 'when the new user info is saved' do
+      before do
+        get :login_dot_gov
+      end
+
+      it 'saves the email' do
+        expect(user.email).to(eq 'test@gsa.gov')
+      end
+      it 'saves the uid' do
+        expect(user.uid).to(eq '12345')
+      end
+    end
+
+    context 'when the existing user with no uid info is saved' do
+      before do
+        mock_user_auth('12345', 'user_with_out_uid@fixtures.org')
```

Comment:
This mock is not being used. In order to use this in the spec, you'd need to add it to the request environment before the login occurs:
```ruby
request.env['omniauth.auth'] = mock_user_auth('12345', 'user_with_out_uid@fixtures.org')
```
I would actually put the auth object into a `let` block, so that it can be set for each context. Example:
```ruby
let(:auth) { mock_user_auth('12345', 'user_with_out_uid@fixtures.org') }

before do
  request.env['omniauth.auth'] = auth
  get :login_dot_gov
end
```

The specs in this context are still using the auth object for test@gsa.gov. They are passing erroneously, due to other issues (see below). I realize these specs were written after the controller code, but TDD would help prevent false positives in the specs.